src/Module.ts

```diff
@@ -708,4 +725,13 @@ export default class Module {
 		warning.id = this.id;
 		this.graph.warn(warning);
 	}
+
+	shimMissingExport(name: string) {
+		const shim = new ShimVariable(name === 'default' ? this.basename() : name);
+		this.exports[name] = {
+			localName: name,
+			shim
+		};
```

Comment:
Would it be possible for `ShimVariable` to inject itself into `originals` here on the module itself so that variable lookups play out correctly? Or does that introduce too many cross-cutting issues?

Reply:
I thought about it, but I couldn't figure out how to get access to the `NamespaceVariable.originals` from earlier in the traceExport call stack at this point of shimming.

Reply:
Let me know if you have further questions on this one.

Reply:
I was able to remove the NamespaceVariable.originals workaround completely. Something I changed must have made it obsolete.